scala/src/main/scala/com/quby/demo/databricks/storage/Repository.scala

```diff
@@ -3,11 +3,59 @@ package com.quby.demo.databricks.storage
 import com.quby.demo.databricks.schema.{ElectricityPowerSchema, RawSchema}
 import org.apache.spark.sql.Dataset
 
+/***
+  * Repository is the interface we use to access data from the storage.
+  *
+  * The two implementations are:
+  *  * SampleRepository -> Use this one for generating sample data on-the-fly
+  *  * DeltaRepository -> Use this one for accessing data stored on Databricks Delta
+  */
 trait Repository {
-  def raw(dateFrom: String, dateTo: String): Dataset[RawSchema]
 
-  def electricityPower(dateFrom: String, dateTo: String): Dataset[ElectricityPowerSchema]
+  /***
+    * The raw table contains all the IoT signals collected from the devices stored with RawSchema schema.
+    * The table is partitioned by utcDate and variableName.
+    *
+    * FAQ: Why using a single table in long format for all signals?
```

Comment:
This FAQ information is very interesting in general as well. Maybe in the future also add them to a general FAQ in readme?

Reply:
Yes, I will do that as well in a separate MR.